### PR TITLE
[MLIR][Doc] Remove LLVM dialect typed pointer documentation

### PR DESCRIPTION
**Pull request by Dinistro**
This commit removes all references to typed pointers. Typed pointers have been deprecated for a while now and they will be removed in a followup.

Related PSA: https://discourse.llvm.org/t/psa-removal-of-typed-pointers-from-the-llvm-dialect/74502

### CODE REVIEW

File: mlir/include/mlir/Conversion/LLVMCommon/Pattern.h [hunk 1]
```diff
@@ -89,10 +89,10 @@ class ConvertToLLVMPattern : public ConversionPattern {
   /// `strides[1]` = llvm.mlir.constant(1 : index) : i64
   /// `strides[0]` = `sizes[0]`
   /// %size        = llvm.mul `sizes[0]`, `sizes[1]` : i64
-  /// %nullptr     = llvm.mlir.zero : !llvm.ptr<f32>
+  /// %nullptr     = llvm.mlir.zero : !llvm.ptr
   /// %gep         = llvm.getelementptr %nullptr[%size]
-  ///                  : (!llvm.ptr<f32>, i64) -> !llvm.ptr<f32>
-  /// `sizeBytes`  = llvm.ptrtoint %gep : !llvm.ptr<f32> to i64
+  ///                  : (!llvm.ptr, i64) -> !llvm.ptr, f32
+  /// `sizeBytes`  = llvm.ptrtoint %gep : !llvm.ptr, f32
```
Comment (zero9178) [inline mlir/include/mlir/Conversion/LLVMCommon/Pattern.h:95]:
```suggestion
  /// `sizeBytes`  = llvm.ptrtoint %gep : !llvm.ptr to i64
```
Looks like a replace-all action went wrong here

File: mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td [hunk 2]
```diff
@@ -1174,12 +1174,12 @@ def LLVM_GlobalOp : LLVM_Op<"mlir.global",
...
-    llvm.mlir.global constant @int_gep() : !llvm.ptr<i32> {
-      %0 = llvm.mlir.addressof @g2 : !llvm.ptr<i32>
+    llvm.mlir.global constant @int_gep() : !llvm.ptr {
+      %0 = llvm.mlir.addressof @g2 : !llvm.ptr
       %1 = llvm.mlir.constant(2 : i32) : i32
       %2 = llvm.getelementptr %0[%1]
-         : (!llvm.ptr<i32>, i32) -> !llvm.ptr<i32>
-      llvm.return %2 : !llvm.ptr<i32>
+         : (!llvm.ptr, i32) -> !llvm.ptr, i32
+      llvm.return %2 : !llvm.ptri32>
```
Comment (zero9178) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td:1182]:
```suggestion
      llvm.return %2 : !llvm.ptr
```

**Review by zero9178 — APPROVED** [review]
LGTM % two comments

**Comment by Dinistro** [comment]
Thanks for the review :smile:

**Pull request merged**